Convert to Polymer 3.0

dpikhulya opened the pull request:
The element (with tests and the demo) was converted to the Polymer 3.0 format (using npm instead of bower, and JavaScript modules instead of HTML imports).

Conversion was made with [polymer-modulizer](https://github.com/Polymer/polymer-modulizer), and a subsequent correction of the auto-translated code.

dpikhulya committed:
Update .gitignore to ignore Idea files, and bower_components directory

dpikhulya committed:
Convert the element to Polymer 3.0

dpikhulya committed:
Fix the Travis build issues

dpikhulya committed:
Revise the converted tests, dependencies, build

dpikhulya committed:
Correct the `name` field in package.json

dpikhulya committed:
Correct the style of comments (copyrights, JSDocs), template indentation

dpikhulya committed:
Revise code layout in element's test

dpikhulya committed:
Remove bower_components from .gitignore

dpikhulya committed:
Simplify the auto-translated demo code

dpikhulya committed:
Remove bower.json (the package is expected to be used with npm now)

dpikhulya committed:
Simplify the auto-translated test code

dpikhulya committed:
Remove `polymer install` command from .travis.yml

...as `npm install` is run by Travis anyway

armiol commented on spine-badge.js:
```diff
@@ -0,0 +1,95 @@
+ *
+ * @demo demo/index.html
+ *
+ * TODO:2017-04-20:dmitry.pikhulya: we might need to make this element compatible with paper-badge
```
It is a good time to maybe drop this todo item. It's been a year. It should be either a task/issue in GitHub, or be gone.

dpikhulya replied:
Removed the todo notice. The paper-elements collection is going to be [replaced](https://www.polymer-project.org/blog/2018-05-02-roadmap-update#elements), so we should probably not try to imitate it.

armiol reviewed (approved):
@dpikhulya LGTM with a comment to address.

dpikhulya committed:
Remove the TODO comment on making it a superset of `paper-badge`

The reasons:
- The paper-elements collection is going to be replaced with a different
  library (see
  [here](https://www.polymer-project.org/blog/2018-05-02-roadmap-update#elements)),
  so it's probably not worth trying to be close to an obsolete library.
- `spine-badge`, by its purpose, will have a different layout for a long
  content (an elongated rounded rectangle instead of a circle in
  `paper-badge`), so it won't be an entirely backwards-compatible
  replacement for `paper-badge` anyway.
